### ISSUE TITLE
Removing a disconnected remote cluster breaks web UI loading

### ISSUE THREAD

**Issue by awly** [gravitational/teleport#4317]
### Description

**What happened**:

Customer removed a leaf cluster and needed to clean it up from the root cluster's web UI.
They ran: `tctl rm rc/leaf-name`.

Afterwards, the root web UI failed to load the home page with: `no active peers found for leaf-name`.

**What you expected to happen**:

The root web UI should load without the leaf cluster in the list.

**Workaround**:

Restart the proxy.

**How to reproduce it (as minimally and precisely as possible)**:

- create a root/leaf cluster setup
- stop the leaf cluster
- run `tctl rm rc/leaf-name` on the root auth server
- reload the root web UI

Possibly related to https://github.com/gravitational/teleport/issues/4183

**Comment by benarent** [comment]
Does the same thing happen with ` tsh clusters` ?

**Comment by awly** [comment]
@benarent haven't had a chance to repro yet.

**Comment by quodlibetor** [comment]
This just happened to me, and `tsh` reports the cluster as offline instead of nonexistent:

```
$ tsh clusters
Cluster Name Status
------------ -------
tsh          online
dev          online
test         offline
```

**Comment by andrejtokarcik** [comment]
I'm unable to reproduce either the UI failure or the offline `tsh clusters` record. In either case, the remote cluster is immediately removed from the table. Is this still occurring on your end?

**Comment by klizhentas** [comment]
@andrejtokarcik we got a customer who just reproduced it

**Comment by andrejtokarcik** [comment]
What Teleport version do they use? It's also possible the reproduction steps above are oversimplified.

**Comment by russjones** [comment]
Since we have not been able to reproduce this, I am going to close this issue for now. Highly likely this was a symptom of early cache reads that was resolved in #4183.

If it continues to occur we can reopen or create another ticket to track this issue.
